DeezyMatch/tests/test_utils.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+import pytest
+
+from DeezyMatch import utils
+
+def test_string_split():
+
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["char"], 
+            "prefix_suffix": ["|", "|"]}
+    assert utils.string_split(**kwds) == ['|', 'p', 'y', ' ', '0', '0', '1', ' ', '$', ' ', ' ', '|']
+    
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["char"],  
+            "prefix_suffix": ["|", None]}
+    assert utils.string_split(**kwds) == ['|', 'p', 'y', ' ', '0', '0', '1', ' ', '$', ' ', ' ']
```

Comment:
Hi @kasra-hosseini,

I just realised that in this case, the value of `prefix_suffix` is read from the input file as a list of strings (i.e. `['|', 'None']`) instead of `['|', None]`.

Therefore, in the following bit of code:
```
x_bounded = copy.deepcopy(x)
if isinstance(prefix_suffix, collections.abc.Sequence) and len(prefix_suffix) == 2:
        prefix = prefix_suffix[0] if isinstance(prefix_suffix[0], str) else ""
        suffix = prefix_suffix[1] if isinstance(prefix_suffix[1], str) else ""
        x_bounded = prefix + x + suffix
```

What happens is that we're actually appending the string "None" as a suffix. E.g. string "Hey" is tokenized as `['|', 'H', 'E', 'Y', 'N', 'o', 'n', 'e']`.

It escaped our notice because the function in the test does read it as a None!

Reply:
Thanks again for this. I added notes here: https://github.com/Living-with-machines/DeezyMatch/pull/114/commits/14a9a1feeb44d8e4a6588cf7591ebb5dd3d7369c and updated the tests accordingly.